src/bot.ml

```diff
@@ -670,452 +89,101 @@ let callback _conn req body =
-                                 comment_info.author e)
-                            ~id:pr.id ))
-            |> Lwt.async ;
+            (fun () -> merge_pull_request ~comment_info ~bot_info) |> Lwt.async ;
             Server.respond_string ~status:`OK
               ~body:(f "Received a request to merge the PR.")
               () )
           else
             Server.respond_string ~status:`OK
               ~body:(f "Unhandled comment: %s." body)
               ()
-      | Ok (_, GitHub_subscriptions.NoOp s) ->
+      | Ok (_, UnsupportedEvent s) ->
           Server.respond_string ~status:`OK ~body:(f "No action taken: %s" s) ()
       | Ok _ ->
           Server.respond_string ~status:`OK
             ~body:"No action taken: event or action is not yet supported." ()
-      | Error s ->
+      | Error ("Webhook signed but with wrong signature." as e) ->
+          Stdio.print_string e ;
+          Server.respond_string ~status:(Code.status_of_code 401)
+            ~body:(f "Error: %s" e) ()
+      | Error e ->
           Server.respond_string ~status:(Code.status_of_code 400)
-            ~body:(f "Error: %s" s) () )
+            ~body:(f "Error: %s" e) () )
   | "/coq-bug-minimizer" ->
-      body
-      >>= fun body ->
-      if string_match ~regexp:"\\([^\n]+\\)\n\\([^\r]*\\)" body then
-        let stamp = Str.matched_group 1 body in
-        let message = Str.matched_group 2 body in
-        match Str.split (Str.regexp " ") stamp with
-        | [id; author; repo_name; branch_name] ->
-            (fun () ->
-              GitHub_mutations.post_comment ~bot_info ~id
-                ~message:(f "@%s, %s" author message)
-              <&> ( execute_cmd
-                      (f
-                         "git push https://%s:%s@github.com/%s.git --delete \
-                          '%s'"
-                         bot_name github_access_token repo_name branch_name)
-                  >>= function
-                  | Ok () ->
-                      Lwt.return ()
-                  | Error f ->
-                      Lwt_io.printf "Error: %s" f ))
-            |> Lwt.async ;
-            Server.respond_string ~status:`OK ~body:"" ()
-        | _ ->
-            Server.respond_string ~status:(`Code 400) ~body:"Bad request" ()
-      else Server.respond_string ~status:(`Code 400) ~body:"Bad request" ()
+      body >>= fun body -> coq_bug_minimizer_results_action body ~bot_info
   | _ ->
       Server.respond_not_found ()
 
-let server =
-  (fun () ->
-    Lwt_io.printf "Initializing repository...\n"
-    <&> ( "git init --bare"
-        |&& f "git config user.email \"%s\"" bot_email
-        |&& f "git config user.name \"%s\"" bot_name
-        |> execute_cmd >|= ignore ))
-  |> Lwt.async ;
+let launch =
+  init_git_bare_repository ~bot_info
+  >>= fun () ->
+  Stdio.printf "Starting server.\n" ;
   let mode = `TCP (`Port port) in
   Server.create ~mode (Server.make ~callback ())
```

Comment:
I'm a bit worried that the change of strategy here (wait until the git repository is initialized before creating the server) is going to generate even more timeouts.

Reply:
Now we initialize the git bare repository before each action calling git commands under the hood.